model-server/src/main/kotlin/org/modelix/model/server/store/StoreClientAdapter.kt

```diff
@@ -76,8 +76,42 @@ abstract class StoreClientAdapter(val client: IsolatingStore) : IStoreClient {
         return client.generateId(key.withRepoScope())
     }
 
-    override fun <T> runTransaction(body: () -> T): T {
-        return client.runTransaction(body)
+    override fun getTransactionManager(): ITransactionManager {
+        return client.getTransactionManager()
+    }
+
+    override fun getImmutableStore(): IImmutableStore<String> {
+        val immutableStore = client.getImmutableStore()
+        return object : IImmutableStore<String> {
+            override fun getAll(keys: Set<String>): Map<String, String> {
+                val fromRepository = immutableStore.getAll(keys.map { it.withRepoScope() }.toSet()).mapKeys { it.key.key }
+                if (getRepositoryId() == null) return fromRepository
+
+                // Existing databases may have objects stored without information about the repository.
+                // Try to load these legacy entries.
+                val missingKeys = fromRepository.entries.asSequence().filter { it.value == null }.map {
```

Comment:
Unfortunately, the statement is correct, which means the immutable store is supposed to return null values instead of throwing an exception.

Reply:
`val fromRepository = immutableStore.getAll(keys.map { it.withRepoScope() }.toSet()).mapKeys { it.key.key }` is `Map<String, String>`. `fromRepository` was already queried from an `IImmutableStore`.

Reply:
What I mean is that `fromRepository` should be `Map<String, String?>` exactly because of this filter operation. The filter is needed and I had to change the return type of `IImmutableStore.getAll`.